sklearn/preprocessing/_encoders.py

```diff
@@ -604,6 +608,18 @@ def fit(self, X, y=None):
         self
 
         """
+
+        checkd_X = check_array(X, dtype=None)
+        contain_str = checkd_X.dtype == object or np.issubdtype(
+            checkd_X.dtype, np.str_)
+        if self.categories == 'auto' and contain_str:
+            warnings.warn(
+                "From version 0.24, OrdinalEncoder's categories='auto' "
+                "setting will not work with string-valued features. "
+                "categories='sort' or an explicit "
+                "category order will be required.",
+                DeprecationWarning, stacklevel=1)
```

Comment:
We should now raise `FutureWarning` as per #15080.

Reply:
Sure. Changed it. Thanks